test/reactionsystem_structure/higher_order_reactions.jl

```diff
@@ -36,49 +40,75 @@ let
+# not fully reproducible.
+let 
+    @test_broken false
+    # Declares a JumpSystem manually. Would have used Catalyst again, but `binomial` still errors when
+    # called on symbolics. For reference, here is the network as it would be created using Catalyst. 
+
+    # higher_order_network_alt2 = @reaction_network begin
+    #     p, ∅ ⟼ X1
+    #     r1 * binomial(X1, 2), 2X1 ⟾ 3X2
+    #     mm(X1, r2, K) * binomial(X2, 3), 3X2 ⟾ X3 + 2X4
+    #     r3 * binomial(X3, 1) * binomial(X4, 2), X3 + 2X4 ⟾ 3X5 + 3X6
+    #     r4 * X2 * binomial(X5, 3) * binomial(X6, 3), 3X5 + 3X6 ⟾ 3X5 + 2X7 + 4X8
+    #     r5 * binomial(X5, 3) * binomial(X7, 2) * binomial(X8, 4), 3X5 + 2X7 + 4X8 ⟾ 10X9
+    #     r6 * binomial(X9, 10), 10X9 ⟾ X10
+    #     d * binomial(X10, 2), 2X10 ⟾ ∅
+    # end
 
-    for factor in [1e-1, 1e0]
-        u0 = rand(rng, 1:Int64(factor * 100), length(get_unknowns(higher_order_network_1)))
-        p = factor * rand(rng, length(get_ps(higher_order_network_3)))
-        prob1 = JumpProblem(higher_order_network_1,
-                            DiscreteProblem(higher_order_network_1, u0, (0.0, 1000.0), p),
-                            Direct())
-        sol1 = solve(prob1, SSAStepper())
-        prob2 = JumpProblem(higher_order_network_3,
-                            DiscreteProblem(higher_order_network_3, u0, (0.0, 1000.0), p),
-                            Direct())
-        sol2 = solve(prob2, SSAStepper())
-        for i in 1:length(u0)
-            vals1 = getindex.(sol1.u, i)
-            vals2 = getindex.(sol1.u, i)
-            (mean(vals2) > 0.001) && @test 0.8 < mean(vals1) / mean(vals2) < 1.25
-            (std(vals2) > 0.001) && @test 0.8 < std(vals1) / std(vals2) < 1.25
-        end
+    rate1(u, p, t) = p[1]
+    rate2(u, p, t) = p[2] * binomial(u[1], 2)
+    rate3(u, p, t) = mm(u[1], p[3], p[4]) * binomial(u[2], 3)
+    rate4(u, p, t) = p[5] * binomial(u[3], 1) * binomial(u[4], 2)
+    rate5(u, p, t) = p[6] * u[2] * binomial(u[5], 3) * binomial(u[6], 3)
+    rate6(u, p, t) = p[7] * binomial(u[5], 3) * binomial(u[7], 2) * binomial(u[8], 4)
+    rate7(u, p, t) = p[8] * binomial(u[9], 10)
+    rate8(u, p, t) = p[9] * binomial(u[10], 2)
+
+    affect1!(int) = (int.u[1] += 1)
+    affect2!(int) = (int.u[1] -= 2; int.u[2] += 3;)
+    affect3!(int) = (int.u[2] -= 3; int.u[3] += 1; int.u[4] += 2;)
+    affect4!(int) = (int.u[3] -= 1; int.u[4] -= 2; int.u[5] += 3; int.u[6] += 3;)
+    affect5!(int) = (int.u[5] -= 3; int.u[6] -= 3; int.u[5] += 3; int.u[7] += 2; int.u[8] += 4;)
+    affect6!(int) = (int.u[5] -= 3; int.u[7] -= 2; int.u[8] -= 4; int.u[9] += 10;)
+    affect7!(int) = (int.u[9] -= 10; int.u[10] += 1;)
+    affect8!(int) = (int.u[10] -= 2;)
+
+    higher_order_network_alt2 = ConstantRateJump.([rate1, rate2, rate3, rate4, rate5, rate6, rate7, rate8], 
+                                [affect1!, affect2!, affect3!, affect4!, affect5!, affect6!, affect7!, affect8!])
```

Comment:
This is a nice test too, so we should get the old Catalyst version fixed and keep your new JumpProcesses version too and compare against both going forward.

Reply:
agreed